tests/pipeline/test_python_component.py

```diff
@@ -237,10 +247,12 @@ def load(self) -> dd.DataFrame:
 
     assert len(pipeline._graph.keys()) == 1
     operation_spec = pipeline._graph["CreateData"]["operation"].operation_spec.to_json()
-    assert json.loads(operation_spec) == {
+    operation_spec_without_image = json.loads(operation_spec)
+    operation_spec_without_image["specification"].pop("image", None)
+
+    assert operation_spec_without_image == {
         "specification": {
             "name": "CreateData",
-            "image": "fondant:latest",
             "description": "python component",
             "consumes": {"additionalProperties": True},
             "produces": {"additionalProperties": True},
```

Comment:
No test for custom image name ?

Reply:
I've updated one of the tests to check the image name. The test itself replicated the logic of the function that generates the image name itself. When we are running the tests with tox we have to replace the image name dynamically. Do you have any idea how we can improve this? I think mocking `sys.version_info` could work. But not sure if this is the best option.

Reply:
See my comment above.

I meant adding a test for when the user uses their own image and then checking if the name is propagated properly.

Reply:
I've extracted the image name generation as fixture and modified a test for checking the custom image. Added a test for checking the extra_requirements warning as well.